core/src/signed_envelope.rs

```diff
@@ -44,15 +44,19 @@ impl SignedEnvelope {
         self.key.verify(&buffer, &self.signature)
     }
 
-    /// Extract the payload of this [`SignedEnvelope`].
+    /// Extract the payload and signing key of this [`SignedEnvelope`].
     ///
     /// You must provide the correct domain-separation string and expected payload type in order to get the payload.
     /// This guards against accidental mis-use of the payload where the signature was created for a different purpose or payload type.
+    ///
+    /// It is the caller's responsibility to check that the signing key is what
+    /// is expected. For example, checking that the signing key is from a
+    /// certain peer.
```

Comment:
How about enforcing this constraint via `#[must_use]`? Something along the lines of the playground below:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1a32dedc2d59c3298dc73a8b48d9d76c

Reply:
Oh excellent! I was wondering if there was a way I could leverage `#[must_use]`. Thanks!

Reply:
As far as I can tell, unfortunately one can not do it on an anonymous tuple. E.g. instead of the additional `struct` I would prefer to do:

``` rust
fn foo() -> Result<(a, #[must_use] b), c> {}
```

Reply:
Yeah that was my first thought, but I hadn't considering using a struct instead.

Reply:
We could do something like this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=92e6fc6cf971873eb590d9ada72f85fa

But I am not sure if it is useful. Assigning `_` will silence the `must_use` warning ...

Reply:
Unfortunately assigning the return value to anything and not using the key value will also satisfy the must_use lint. e.g. `let data = foo().unwrap();`